nestkernel/connection_manager.cpp

```diff
@@ -1311,11 +1336,48 @@ nest::ConnectionManager::get_targets( std::vector< index > sources,
     target_it = targets.begin();
     for ( ; source_it != sources.end(); source_it++, target_it++ )
     {
-      if ( ( *it ).get( *source_it ) != 0 )
+      if ( validate_source_entry_( thread_id, *source_it ) != 0 )
       {
-        validate_pointer( ( *it ).get( *source_it ) )
+        validate_pointer( validate_source_entry_( thread_id, *source_it ) )
           ->get_target_gids( ( *target_it ), thread_id, synapse_model );
       }
     }
   }
 }
+
+void
+nest::ConnectionManager::get_targets( std::vector< index > sources,
+  std::vector< std::vector< index > >& targets,
+  index synapse_model,
+  std::string post_synaptic_element )
+{
+  thread thread_id;
+  std::vector< index >::iterator source_it;
+  std::vector< std::vector< index > >::iterator target_it;
+  targets.resize( sources.size() );
+  for ( std::vector< std::vector< index > >::iterator i = targets.begin();
+        i != targets.end();
+        i++ )
+  {
+    ( *i ).clear();
+  }
+
+  for ( tVSConnector::iterator it = connections_.begin();
```

Comment:
could you turn this into a loop of this sort
```c++
for ( thread tid = 0; tid < connections_.size(); ++tid )
{
  [...]
}
```
or are there reasons to stick to the iterators. I find it quite hard to read. this would also make it easier to introduce a threaded version later.